vuu-ui/packages/vuu-layout/src/layout-persistence/LocalLayoutPersistenceManager.ts

```diff
@@ -1,48 +1,68 @@
-import { Layout, LayoutMetadata, WithId } from "@finos/vuu-shell";
-import { LayoutJSON, LayoutPersistenceManager } from "@finos/vuu-layout";
+import {
+  Layout,
+  LayoutMetadata,
+  LayoutMetadataDto,
+  WithId,
+} from "@finos/vuu-shell";
 import { getLocalEntity, saveLocalEntity } from "@finos/vuu-filters";
-import { getUniqueId } from "@finos/vuu-utils";
+import { formatDate, getUniqueId } from "@finos/vuu-utils";
 
 import { defaultLayout } from "./data";
+import { LayoutPersistenceManager } from "./LayoutPersistenceManager";
+import { LayoutJSON } from "../layout-reducer";
 
 const metadataSaveLocation = "layouts/metadata";
 const layoutsSaveLocation = "layouts/layouts";
 
 export class LocalLayoutPersistenceManager implements LayoutPersistenceManager {
-  createLayout(metadata: Omit<LayoutMetadata, "id">, layout: LayoutJSON): Promise<string> {
-    return new Promise(resolve => {
-      console.log(`Saving layout as ${metadata.name} to group ${metadata.group}...`);
+  createLayout(
+    metadata: LayoutMetadataDto,
+    layout: LayoutJSON
+  ): Promise<LayoutMetadata> {
+    return new Promise((resolve) => {
+      console.log(
+        `Saving layout as ${metadata.name} to group ${metadata.group}...`
+      );
 
-      Promise.all([this.loadLayouts(), this.loadMetadata()])
-        .then(([existingLayouts, existingMetadata]) => {
+      Promise.all([this.loadLayouts(), this.loadMetadata()]).then(
+        ([existingLayouts, existingMetadata]) => {
           const id = getUniqueId();
-          this.appendAndPersist(
+          const newMetadata = {
+            ...metadata,
             id,
-            metadata,
-            layout,
-            existingLayouts,
-            existingMetadata
+            created: formatDate(new Date(), "dd.mm.yyyy"),
+          };
+
+          this.saveLayoutsWithMetadata(
+            [...existingLayouts, { id, json: layout }],
+            [...existingMetadata, newMetadata]
           );
-          resolve(id);
-        });
-    })
+          resolve(newMetadata);
+        }
+      );
+    });
   }
 
   updateLayout(
     id: string,
-    newMetadata: Omit<LayoutMetadata, "id">,
+    newMetadata: LayoutMetadataDto,
     newLayout: LayoutJSON
   ): Promise<void> {
     return new Promise((resolve, reject) => {
       this.validateIds(id)
         .then(() => Promise.all([this.loadLayouts(), this.loadMetadata()]))
         .then(([existingLayouts, existingMetadata]) => {
-          const layouts = existingLayouts.filter(layout => layout.id !== id);
-          const metadata = existingMetadata.filter(metadata => metadata.id !== id);
-          this.appendAndPersist(id, newMetadata, newLayout, layouts, metadata);
+          this.saveLayoutsWithMetadata(
+            existingLayouts.map((layout) =>
+              layout.id === id ? { ...layout, json: newLayout } : layout
```

Comment:
I'd find it easier to read with these arguments extracted into variables, but I like the overall change to mapping rather than filtering/appending.

Reply:
change applied :)